lib/libc/stdlib/malloc/mrs/mrs.c

```diff
@@ -1470,16 +1484,18 @@ mrs_rallocx(void *ptr, size_t size, int flags)
 		return (REAL(rallocx)(ptr, size, flags));
 
 	size_t old_size = cheri_getlen(ptr);
+	void *new_alloc;
```

Comment:
If you're going to bother to hoist new_alloc, hoist both variables to the actual start of the block?